src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,354 @@
+            args.append(self.visit(param))
+        function_signature = ",".join([function_signature, *args])
+        function_signature += ",_domain_, _origin_):"
+        self.body.append(function_signature)
+        self.body.indent()
+
+    def generate_stencil_code(
+        self,
+        stencil: Stencil,
+        block_extents: dict[int, Extent],
+    ):
+        for loop in stencil.vertical_loops:
+            for section in loop.sections:
+                with self.create_k_loop_code(section, loop):
+                    for execution in section.horizontal_executions:
+                        with self.generate_ij_loop(block_extents, execution):
+                            self.visit(execution)
+
+    @contextmanager
+    def create_k_loop_code(self, section: VerticalLoopSection, loop: VerticalLoop) -> Generator:
+        loop_bounds: str = self.visit(section.interval, var="k", direction=loop.loop_order)
+        iterator = "1" if loop.loop_order != LoopOrder.BACKWARD else "-1"
+        loop_code = "for k in range(" + loop_bounds + "," + iterator + "):"
+        self.body.append(loop_code)
+        self.body.indent()
+        yield
+        self.body.dedent()
+
+    @contextmanager
+    def generate_ij_loop(
+        self, block_extents: dict[int, Extent], execution: HorizontalExecution
+    ) -> Generator:
+        extents = block_extents[id(execution)]
+        i_loop = f"for i in range(i_0 + {extents[0][0]} , i_size + {extents[0][1]}):"
+        self.body.append(i_loop)
+        self.body.indent()
+        j_loop = f"for j in range(j_0 + {extents[1][0]} , j_size + {extents[1][1]}):"
+        self.body.append(j_loop)
+        self.body.indent()
+        yield
+        self.body.dedent()
+        self.body.dedent()
+
+    def visit_While(self, while_node: While, **_) -> None:
+        while_condition = self.visit(while_node.cond)
+        while_code = f"while {while_condition}:"
+        self.body.append(while_code)
+        self.body.indent()
+        for statement in while_node.body:
+            self.visit(statement)
+        self.body.dedent()
+
+    def visit_FieldDecl(self, field_decl: FieldDecl, **_) -> str:
+        return str(field_decl.name)
+
+    def visit_AxisBound(self, axis_bound: AxisBound, **kwargs):
+        if axis_bound.level == LevelMarker.START:
+            return f"{kwargs['var']}_0 + {axis_bound.offset}"
+        if axis_bound.level == LevelMarker.END:
+            return f"{kwargs['var']}_size + {axis_bound.offset}"
+
+    def visit_Interval(self, interval: Interval, **kwargs):
+        if kwargs["direction"] == LoopOrder.BACKWARD:
+            return ",".join(
+                [
+                    self.visit(interval.end, **kwargs) + "- 1",
+                    self.visit(interval.start, **kwargs) + "- 1",
+                ]
+            )
+        else:
+            return ",".join(
+                [self.visit(interval.start, **kwargs), self.visit(interval.end, **kwargs)]
+            )
```

Comment:
```suggestion
    def visit_Interval(self, interval: Interval, **kwargs):
        if kwargs["direction"] == LoopOrder.BACKWARD:
            return ",".join(
                [
                    self.visit(interval.end, **kwargs) + "- 1",
                    self.visit(interval.start, **kwargs) + "- 1",
                ]
            )

        return ",".join(
            [self.visit(interval.start, **kwargs), self.visit(interval.end, **kwargs)]
        )
```

No explicit `else` needed after `return` in `if`. Reduces the amount of indent needed. It might even make sense to change to "if direction == FORWARD:" because that case is simpler and the you get less indent levels for the "more complicated" case of `BACKWARD` looping.